packages/astro/src/assets/internal.ts

```diff
@@ -90,6 +90,15 @@ export async function getImage(
 				: options.src,
 	};
 
+	// Clone the `src` object if it's an ESM import so that we don't refer to any properties of the original object
+	// Causing our generate step to think the image is used outside of the image optimization pipeline
+	const clonedSrc = isESMImportedImage(resolvedOptions.src)
+		? // @ts-expect-error - clone is a private, hidden prop
+		  resolvedOptions.src.clone ?? resolvedOptions.src
```

Comment:
I believe you can use an `/** @internal */` comment to have erasable types for `ImageMetadata`. Properties marked with `@internal` won't be included when `.d.ts` files are generated, assuming [`stripInternal`](https://www.typescriptlang.org/tsconfig#stripInternal) is enabled.

Reply:
Oh, I didn't know that, that's neat. Will do that